src/esp/gfx/ShaderManager.h

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#pragma once
+
+#include <Magnum/GL/AbstractShaderProgram.h>
+#include <Magnum/ResourceManager.h>
+
+#include "esp/gfx/LightSetup.h"
+#include "esp/gfx/MaterialData.h"
+
+namespace esp {
+namespace gfx {
+
+using ShaderManager = Magnum::ResourceManager<Magnum::GL::AbstractShaderProgram,
```

Comment:
`Magnum::ResourceManager` is a true singleton (lifetime tied to the program itself, not an instance of the simulator). We currently make assumptions that a true singleton violates -- that all opengl resources are cleared when the simulator is killed. This is why that test is failing.

(We should add an explicit test for being able to construct/destruct the simulator multiple times as that one is checking something specific and won't catch everything involved).

Reply:
@mosra Any thoughts? Really would not like to have to reimplement the `ResourceManager` logic if we could avoid it. Perhaps an option could be added on the Magnum side?

Reply:
After diving into the Magnum code, the singleton functionality of `ResourceManager` is actually deprecated. I turned off the `BUILD_DEPRECATED` build flag (and minor related build issues) and it fixes the issue.

Reply:
@erikwijmans hmm, no, the ResourceManager isn't a *true singleton* -- as with all other globals in Magnum, the application has the full control over its lifetime. I deprecated it for slightly different reasons -- it's hard/impossible to have a truly unique global instance pointer for a template class and because all attempts to mitigate this problems in a generic way only made it worse, I decided to move this responsibility on the users instead.

I'd say the true cause of abort was due to two instances of the resource manager existing at the same time -- is that the case in tests? (It should print that on the error output, if you run `pytest` in a certain way I think.) And if so, could that be avoided, for the time being at least? That check isn't present anymore when you disable `BUILD_DEPRECATED`, but I'd like to keep that flag on since it makes upgrades *a lot* smoother. I'd like to remove this deprecated feature soon, but unfortunately various users still rely on this feature so I can't get rid of it just yet.

Reply:
Agreed, we should be avoiding multiple instances in one process. This should be the case while `Magnum::ResourceManager` is a member of `esp::assets::ResourceManager`. Is it true that we try to instantiate this twice in some existing test?

I'm fine with adding an explicit test for closing and re-initializing the simulator multiple times, this has been an issue in the past for other changes.

Reply:
We are instantiating multiple instances of `Simulator` in the **same process**, which in turn causes two instances of `ResourceManager` in this test:

https://github.com/facebookresearch/habitat-sim/blob/d281a086d7cc24a4269276c21d6c55e3aa6c989f/tests/test_sensors.py#L130-L143

We need to decide if this a requirement or not @erikwijmans @aclegg3.

Reply:
> hmm, no, the ResourceManager isn't a true singleton -- as with all other globals in Magnum, the application has the full control over its lifetime.

@mosra That's great!  I poked around it only a little bit.

I am fine with only allowing one instance of the simulator at a time (this is already a requirement when you have any OpenGL resources).  @matthewjmay can you put a skip on the current test that is failing and some kind of comment/message to take that skip away once @mosra deprecates that officially?

I also added an explicit test for multiple construct/destructs here: https://github.com/facebookresearch/habitat-sim/pull/490

Reply:
Is the simulator instantiating the GLContext as well? Because that one has a similar assert check (allowing only one instance to be "current" at a time), so I wonder why that one isn't triggering also in this case. Or is the GLContext external to the simulator (and then maybe ResourceManager could follow suit) or some extra `makeCurrent()` calls done to it?

Reply:
In the specific case of that test, it isn't.  That test is instating the simulator for a "blind" agent (which shouldn't load opengl things since its blind and we don't need to render for it).

Reply:
I think we have an explicit `try_create` for the context:
https://github.com/facebookresearch/habitat-sim/blob/d281a086d7cc24a4269276c21d6c55e3aa6c989f/src/esp/gfx/WindowlessContext.cpp#L256-L261